src/box/iproto.cc

```diff
@@ -138,6 +138,10 @@ struct iproto_thread {
 	 * List of stopped connections
```

Comment:
In the commit message you said you added a list of all connections, but that is not the only thing you did. You also started doing something with the shutdown flag. Does not look atomic.